apps/govquests-frontend/src/domains/notifications/components/NotificationToaster.tsx

```diff
@@ -0,0 +1,44 @@
+import { Toaster } from "@/components/ui/toaster";
+import { useToast } from "@/hooks/use-toast";
+import { useNotifications } from "../hooks/useNotifications";
+import { useEffect, useRef } from "react";
+
+const NotificationToaster = () => {
+  const { toast } = useToast();
+
+  const { data } = useNotifications();
+
+  const notifications = data?.pages.flatMap((page) => page.notifications.edges);
+  const prevNotifications: typeof notifications = usePrevious(notifications);
+
+  const toastNewNotification = () => {
+    if (!!prevNotifications && notifications !== prevNotifications) {
+      const newNotifications = notifications.filter(
+        (notification) =>
+          !prevNotifications.find(
+            (value) => value.node.id == notification.node.id,
+          ),
+      );
+      newNotifications.forEach((notification) => {
+        if (notification?.node?.notificationType === "quest_completed") {
+          toast({ title: notification?.node?.content });
+        }
+      });
+    }
+  };
+
+  useEffect(toastNewNotification, [notifications]);
+
+  return <Toaster />;
+};
```

Comment:
maybe (most definitely for sure) what I'll suggest now would be a bit overkill but perhaps a good learning opportunity for design patterns. right now, this component is doing multiple things:

- comparing prev with current notifications to filter new ones
- implementing a strategy pattern for "quest_completed" to only toast it
- handling side effects with useEffect
- dealing with the UI toast stuff

we could split this up using some neat patterns:

1. move the diff logic into its own hook (useNotificationDiff):
   - just handles the prev/current comparison
   - makes it easier to test just that piece
   - cleaner than our inline filtering

2. use a factory pattern for handling different notification types:
   - NotificationHandlerFactory creates handlers for each type
   - each notification type gets its own class
   - super easy to add new types later
   - no need to touch existing code when adding stuff

3. strategy pattern for different notification behaviors:
   - handlers all follow same interface
   - each type can do its own thing
   - easy to swap behaviors around

what this would be nice for, imo:
- testing (can test each piece separately)
- adding new notification types later
- keeping things organized as we grow

lmk if you want to see what this might look like in code! i can show a quick example of how we'd structure it